docs: fix claims to roles description in access controls reference (#25036)
gravitational/teleport: docs/pages/includes/role-spec.mdx

      # the `claims_to_roles` mapping works the same as it does in
      # the OIDC connector, with the added benefit that the roles being mapped to

>       # can also be matchers. this example leverages Teleport's regular expression
>       # support that allows for dynamic mapping from claims. the below mapping
>       # says that users with claims that match with `projects: project-(.*)` can request
>       # roles that match with '$1-admin' with the first match group.

Reviewer: ```suggestion
      # can also be matchers. 
      #
      # This example leverages Teleport's regular expression support, which allows 
      # for dynamic mapping from claims. The below mapping says that users with
      # claims that match "projects: project-(.*)" can request roles that match 
      # "$1-admin", where "$1" is the first capture group in the 
      # regular expression.
```

      # the `claims_to_roles` mapping works the same as it does in
      # the OIDC connector, with the added benefit that the roles being mapped to
      # can also be matchers. 
      #
      # This example leverages Teleport's regular expression support, which allows 
      # for dynamic mapping from claims. The below mapping says that users with
      # claims that match "projects: project-(.*)" can request roles that match 
      # "$1-admin", where "$1" is the first capture group in the 
      # regular expression.

>       # Example: `projects: project-foo` claim allows requesting `foo-admin` role

Reviewer: ```suggestion
      # Example: the "projects: project-foo" claim allows a user to request the
      # "foo-admin" role
```

      # the `claims_to_roles` mapping works the same as it does in
      # the OIDC connector, with the added benefit that the roles being mapped to
      # can also be matchers. 
      #
      # This example leverages Teleport's regular expression support, which allows 
      # for dynamic mapping from claims. The below mapping says that users with
      # claims that match "projects: project-(.*)" can request roles that match 
      # "$1-admin", where "$1" is the first capture group in the 
      # regular expression.

>       # Example: the "projects: project-foo" claim allows a user to request the

Reviewer 2: To match the value below, would it be:

```suggestion
      # Example: the "projects: product-foo" claim allows a user to request the
```
?

Author: Thanks @alexfornuto , yes corrected.

Author: Please approve if that's good to go @alexfornuto

      # the `claims_to_roles` mapping works the same as it does in
      # the OIDC connector, with the added benefit that the roles being mapped to
      # can also be matchers. the below mapping says that users with
      # claims that match to `projects: project-(.*)` can request

>       # roles that match with '$1-admin' with the first match group.

Reviewer 3: Is there a difference between "match to" and "match with"? 

If not, I would use consistent language. 

Probably also worth mentioning that these are regular expressions too.

Author: thanks, update
